src/app/features/rpc-transaction-request/stacks/contract-deploy-details.layout.tsx

```diff
@@ -0,0 +1,52 @@
+import { SharedComponentsSelectors } from '@tests/selectors/shared-component.selectors';
+import { Stack, styled } from 'leather-styles/jsx';
+
+import { StacksUnsignedContractDeployOptions, TransactionTypes } from '@leather.io/stacks';
+import { AddressDisplayer, Approver, Highlighter } from '@leather.io/ui';
+
+import { Prism } from '@app/common/clarity-prism';
+
+interface ContractDeployDetailsLayoutProps {
+  address: string;
+  txOptions: StacksUnsignedContractDeployOptions;
+}
+export function ContractDeployDetailsLayout({
+  address,
+  txOptions,
+}: ContractDeployDetailsLayoutProps) {
+  if (txOptions.txType !== TransactionTypes.ContractDeploy)
```

Comment:
there are utility fns in stacks.js to check type, might be nice to reuse

Reply:
Yeah, I think you are right ...I believe I had it that way before refactoring, thx.

Reply:
You actually have to pass in the unsignedTx payload in those fns, so I don't think will work when passing just the options. We only create the unsignedTx right at the end `onApprove`.